src/InsertTagHandler.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of terminal42/contao-inserttags.
+ *
+ * (c) terminal42
+ *
+ * @license MIT
+ */
+
+namespace Terminal42\InsertTagsBundle;
+
+use Contao\CoreBundle\InsertTag\InsertTagParser;
+use Contao\CoreBundle\String\SimpleTokenParser;
+use Contao\Database;
+use Contao\FrontendUser;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class InsertTagHandler
+{
+    public function __construct(
+        private Connection $connection,
+        private InsertTagParser $insertTagParser,
+        private RequestStack $requestStack,
+        private SimpleTokenParser $simpleTokenParser,
+        private TokenStorageInterface $tokenStorage,
+    ) {}
+
+    /**
+     * Parse the insert tag.
+     */
+    public function parseInsertTag(string $tag): ?string
+    {
+        $chunks = explode('::', $tag);
+
+        if ('custom' !== $chunks[0]) {
+            return null;
+        }
+
+        $record = $this->connection->fetchAssociative('SELECT * FROM tl_inserttags WHERE tag=?', [$chunks[1]]);
+
+        if ($record === false) {
+            return null;
+        }
+
+        // Return the "replacement not" if tag is found but does not validate
+        if (!$this->isTagValid($record)) {
+            return $record['replacementNot'];
+        }
+
+        $tokens = [
+            'page' => $this->getPageModel(),
+            'request' => $this->requestStack->getCurrentRequest(),
+            'tag' => (object) array_slice($chunks, 2),
+            'user' => $this->getFrontendUser(),
```

Comment:
```suggestion
            'member' => $this->getFrontendUser(),
```

I know it's a user too but I guess it's good for our back end users to explicitly use `member` so it's clear that it's FE :D